Player.gd

```diff
@@ -48,21 +48,55 @@ func get_player_input():
 			$AnimationPlayer.play("Attack")
 			$Networking.sync_character_animation = "Attack"
 			
-			rpc_id(1, "spawn_projectile")
+			rpc_id(1, "spawn_projectile_server")
+
 @rpc(any_peer)
-func spawn_projectile():
+func spawn_projectile_server():
+	# TODO consider adding the path to the attack you want.
+	# That way different attacks could be triggered by just changing one argument :)
+	
+	if not multiplayer.is_server():
+		print('Client ' + multiplayer.get_remote_sender_id() + ' told me to shoot. I should report them for cheating!')
+		return
+	# Validate the correct person has triggered this RPC
+	if str(name).to_int() != multiplayer.get_remote_sender_id():
+		print('Client ' + multiplayer.get_remote_sender_id() + ' is trying to have player ' + str(name) + ' shoot. They are cheating!')
+		return
+	
+	# Validate that the player is allowed to attack
+	# This should be done here on the server to prevent cheating.
+	# It should also be done client-side, to save us an RPC call for good clients :)
+	if not $AttackDelay.is_stopped():
+		return
+
+	var pos : Vector2
+	var impulse : Vector2
+	if $AnimatedSprite2D.flip_h:
+		pos = position + Vector2(-5,0)
+		impulse = Vector2(-200,0)
+	else:
+		pos = position + Vector2(20,0)
+		impulse = Vector2(200,0)
+	
+	# Call all clients to make them spawn the projectile. This also runs for us (the server)
+	rpc('spawn_projectile_clients', pos, impulse, $AnimatedSprite2D.flip_h)
+
+@rpc(call_local)
```

Comment:
call_local is a magic keyword that means when the server calls rpc('spawn_projectile_clients'...) it also executes locally on the server.

Note that we're not using any_peer, so godot will only let the multiplayer authority call spawn_projectile_clients. And the multiplayer authority for a Player node is always the server :)